go-controller/pkg/cni/cnishim.go

```diff
@@ -208,67 +207,68 @@ func (p *Plugin) CmdCheck(args *skel.CmdArgs) error {
 	}()
 
 	// read the config stdin args
-	conf, err := config.ReadCNIConfig(args.StdinData)
-	if err != nil {
-		return fmt.Errorf("invalid stdin args")
-	}
-	setupLogging(conf)
-	if gtet, err := version.GreaterThanOrEqualTo(conf.CNIVersion, "0.4.0"); err != nil {
-		klog.Errorf(err.Error())
-		return types.NewError(types.ErrDecodingFailure, err.Error(), "")
-	} else if !gtet {
-		err = types.NewError(types.ErrIncompatibleCNIVersion, "config version does not allow CHECK", "")
-		klog.Errorf(err.Error())
+	// Note: in order to validate that RawPrevResult is not nil, we don't use config.ReadCNIConfig(args.StdinData)
+	conf := &ovntypes.NetConf{}
+	if err = json.Unmarshal(args.StdinData, conf); err != nil {
 		return err
 	}
-
 	// Parse previous result.
 	if conf.RawPrevResult == nil {
 		return fmt.Errorf("required prevResult missing")
 	}
-	if err := version.ParsePrevResult(&conf.NetConf); err != nil {
+	if err = version.ParsePrevResult(&conf.NetConf); err != nil {
 		return err
 	}
-
-	netns, err := ns.GetNS(args.Netns)
-	if err != nil {
-		return fmt.Errorf("failed to open netns %q: %v", args.Netns, err)
-	}
-	defer netns.Close()
-
-	if conf.IPAM.Type != "" {
-		// run the IPAM plugin and get back the config to apply
-		err = ipam.ExecCheck(conf.IPAM.Type, args.StdinData)
+	setupLogging(conf)
+	defer func() {
 		if err != nil {
-			return err
+			klog.Errorf(err.Error())
 		}
-	}
+	}()
 
-	result, err := current.NewResultFromResult(conf.PrevResult)
-	if err != nil {
+	gtet := false
+	if gtet, err = version.GreaterThanOrEqualTo(conf.CNIVersion, "0.4.0"); err != nil {
+		return types.NewError(types.ErrDecodingFailure, err.Error(), "")
+	} else if !gtet {
+		err = types.NewError(types.ErrIncompatibleCNIVersion, "config version does not allow CHECK", "")
 		return err
 	}
 
-	var contMap current.Interface
-	for _, intf := range result.Interfaces {
-		if args.IfName == intf.Name {
-			if args.Netns == intf.Sandbox {
-				contMap = *intf
-				continue
+	result, errR := current.NewResultFromResult(conf.PrevResult)
+	if errR != nil {
+		err = errR
+		return err
+	}
+	klog.Infof("PrevResult is %+v", result)
+
+	var intf current.Interface
+	for _, inf := range result.Interfaces {
+		if args.IfName == inf.Name {
+			if args.Netns == inf.Sandbox {
+				intf = *inf
+				break
 			}
 		}
 	}
 	// The namespace must be the same as what was configured
-	if args.Netns != contMap.Sandbox {
-		return fmt.Errorf("sandbox in prevResult %s doesn't match configured netns: %s",
-			contMap.Sandbox, args.Netns)
+	if args.Netns != intf.Sandbox {
+		err = fmt.Errorf("sandbox in prevResult %s doesn't match configured netns: %s",
+			intf.Sandbox, args.Netns)
+		return err
+	}
+
+	netns, errNs := ns.GetNS(args.Netns)
+	if errNs != nil {
+		err = fmt.Errorf("failed to open netns %q: %v", args.Netns, errNs)
```

Comment:
please fixup this commit into previous commits

Reply:
Done